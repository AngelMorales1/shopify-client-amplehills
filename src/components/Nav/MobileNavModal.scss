@import 'settings';

.MobileNavModal {
  color: color('peach');
  pointer-events: none;
  opacity: 0;

  &--active {
    pointer-events: all;
    opacity: 1;
  }
}
