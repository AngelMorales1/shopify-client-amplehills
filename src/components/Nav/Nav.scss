.left-side {
  > a:first-of-type {
    margin-left: 0;
  }
}
