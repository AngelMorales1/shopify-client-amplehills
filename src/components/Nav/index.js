import React, { Component, Fragment } from 'react';
import { connect } from 'react-redux';
import { bindActionCreators } from 'redux';
import totalItems from 'state/selectors/totalItems';
import {
  openMiniCart,
  closeMiniCart
} from 'state/actions/ui/miniCartUIActions';

import PropTypes from 'prop-types';
import get from 'utils/get';
import cx from 'classnames';
import Global from 'constants/Global';
import imageModel from 'models/imageModel';

import { NavLink } from 'react-router-dom';
import { Image, Button } from 'components/base';
import styles from './Nav.scss';

class Nav extends Component {
  state = {
    currentBreakpoint: Global.breakpoints.medium.label
  };

  componentDidMount() {
    window.addEventListener('resize', this.updateWindow);
    this.updateWindow();
  }

  updateWindow = () => {
    const { small, medium } = Global.breakpoints;
    const currentBreakpoint =
      window.innerWidth <= medium.lowerbound ? small.label : medium.label;

    if (this.state.currentBreakpoint !== currentBreakpoint)
      this.setState({ currentBreakpoint });
  };

  toggleMiniCart = () => {
    const {
      miniCartIsOpen,
      actions: { openMiniCart, closeMiniCart }
    } = this.props;

    return miniCartIsOpen ? closeMiniCart() : openMiniCart();
  };

  render() {
    const { logo, profileIcon } = this.props;

    return (
      <div
        className={cx(
          styles['Nav'],
          this.props.alertIsActive ? null : 't0',
          'z-nav w100 fixed l0 flex bg-peach text-white items-center transition-fade-in'
        )}
      >
        <div
          className={`col col-4 md-col-5 flex items-center ${
            styles['left-side']
          }`}
        >
          <NavLink
            exact
            to="/locations"
            className="link-text center text-hover"
          >
            Locations
          </NavLink>
          <NavLink
            exact
            to="/contact"
            className="text-hover ml3 link-text center xs-hide sm-hide"
          >
            Contact
          </NavLink>
        </div>
        <div
          className={cx(
            styles['logo-container'],
            'col col-3 md-col-4 h100 flex items-center'
          )}
        >
          <NavLink exact to="/" className="flex justify-center">
            <Image
              className={cx(styles['logo-container__logo'], 'col-12 md-col-10')}
              alt="Click the Ample Hills Logo to return to the homepage"
              src={get(logo, 'fields.file.url', '')}
            />
          </NavLink>
        </div>
        <div className={`col col-5 flex items-center ${styles['right-side']}`}>
          {this.state.currentBreakpoint === 'medium' ? (
            <Fragment>
              <NavLink exact to="/profile" className="mr2 link-text center">
                <Image
                  className="icon"
                  src={get(profileIcon, 'fields.file.url', '')}
                />
              </NavLink>
              <Button
                className="mr2"
                to="/products"
                variant="primary-small"
                color="white-peach"
                label="Shop Online"
                hover="clear-white-border"
              />
            </Fragment>
          ) : (
            <Fragment>
              <NavLink
                exact
                to="/products"
                className="text-hover link-text center"
              >
                Shop
              </NavLink>
              <NavLink exact to="/profile" className="link-text center">
                <Image src="/assets/images/icon-profile.svg" />
              </NavLink>
            </Fragment>
          )}
          <Button
            className={cx(
              styles['Nav__cart-button'],
              'small flex items-center justify-center'
            )}
            variant="circle"
            color="madison-blue"
            to="/cart"
            label={this.props.totalItems.toString()}
            hover="clear-white-border"
          />
        </div>
      </div>
    );
  }
}

Nav.propTypes = {
  actions: PropTypes.shape({
    openMiniCart: PropTypes.func,
    closeMiniCart: PropTypes.func
  }),
  miniCartIsOpen: PropTypes.bool,
  logo: imageModel.propTypes,
  profileIcon: imageModel.propTypes
};

Nav.defaultProps = {
  actions: {
    openMiniCart: () => {},
    closeMiniCart: () => {}
  },
  miniCartIsOpen: false,
  logo: imageModel.default,
  profileIcon: imageModel.default
};

const mapStateToProps = state => {
  return {
    ...state,
    miniCartIsOpen: get(state, 'miniCartUI.miniCartIsOpen'),
    totalItems: totalItems(state)
  };
};

const mapDispatchToProps = dispatch => {
  return {
    actions: bindActionCreators(
      {
        openMiniCart,
        closeMiniCart
      },
      dispatch
    )
  };
};

export default connect(
  mapStateToProps,
  mapDispatchToProps
)(Nav);
