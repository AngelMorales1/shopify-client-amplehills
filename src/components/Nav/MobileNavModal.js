import React from 'react';
import PropTypes from 'prop-types';

import { Button, Image } from 'components/base';
import { NavLink } from 'react-router-dom';
import cx from 'classnames';
import get from 'utils/get';
import styles from './MobileNavModal.scss';

const MobileNavModal = props => {
  const classes = cx(
    'p3 fixed-cover bg-white-wash flex flex-column transition',
    styles['MobileNavModal'],
    {
      [styles['MobileNavModal--active']]: props.isNavModalOpen
    }
  );

  return (
    <div className={classes}>
      <div onClick={props.closeMobileNavModal} className="flex mb3">
        <Button className="mr2" variant="style-none">
          <Image
            className="col-9"
            alt="close icon"
            src="/assets/images/icon-close.svg"
          />
        </Button>
        <NavLink exact to="/search">
          <Image alt="search icon" src="/assets/images/icon-search-peach.svg" />
        </NavLink>
      </div>
      <div
        onClick={props.closeMobileNavModal}
        className="flex flex-column items-start ml3"
      >
        <div className="mb2 link-text-mobile">Locations</div>
        <NavLink exact to="/flavors" className="mb2 link-text-mobile">
          Flavors
        </NavLink>
        <NavLink exact to="/classes-socials" className="mb2 link-text-mobile">
          Classes
        </NavLink>
        <NavLink exact to="/parties" className="mb2 link-text-mobile">
          Parties
        </NavLink>
        <NavLink exact to="/events" className="mb2 link-text-mobile">
          Events
        </NavLink>
        <NavLink exact to="/ourstory" className="mb2 link-text-mobile">
          Our Story
        </NavLink>
        <NavLink exact to="/contact" className="mb3 link-text-mobile">
          Contact Us
        </NavLink>
        <Button to="/products" label="Shop Online" color="peach" />
      </div>
    </div>
  );
};

MobileNavModal.propTypes = {};

MobileNavModal.defaultProps = {};
export default MobileNavModal;
