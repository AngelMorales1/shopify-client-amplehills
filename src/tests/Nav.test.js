import React from 'react';
import { shallow } from 'enzyme';

import Nav from 'components/Nav';

it('renders without crashing', () => {
  const component = shallow(<Nav />);

  expect(component).toMatchSnapshot();
});
