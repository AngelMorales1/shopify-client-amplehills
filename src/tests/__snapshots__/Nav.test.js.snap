// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders without crashing 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Nav />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <div
          className="col col-4 flex justify-start undefined"
>
          <NavLink
                    activeClassName="active"
                    aria-current="page"
                    className="ml4 link-text"
                    exact={true}
                    to="/location"
          >
                    Location
          </NavLink>
          <NavLink
                    activeClassName="active"
                    aria-current="page"
                    className="ml4 link-text"
                    exact={true}
                    to="/contact"
          >
                    Contact
          </NavLink>
</div>,
        <div
          className="col col-4 flex justify-center items-center"
>
          <NavLink
                    activeClassName="active"
                    aria-current="page"
                    className="justify-center"
                    exact={true}
                    to="/"
          >
                    <Image
                              alt="Click the Ample Hills Logo to return to the homepage"
                              bg={false}
                              className="w100"
                              onImgLoad={[Function]}
                              src="/assets/images/ample-hills-logo.svg"
                              style={Object {}}
                    />
          </NavLink>
</div>,
        <div
          className="col col-4 flex items-center justify-end clearfix undefined"
>
          <NavLink
                    activeClassName="active"
                    aria-current="page"
                    className="mr4 link-text"
                    exact={true}
                    to="/profile"
          >
                    Profile
          </NavLink>
          <NavLink
                    activeClassName="active"
                    aria-current="page"
                    className="mr4 link-text"
                    exact={true}
                    to="/collections"
          >
                    Collections
          </NavLink>
          <Button
                    className=""
                    color="peach"
                    disabled={false}
                    fullWidth={false}
                    label="Shop Online"
                    onClick={[Function]}
                    to="/products"
                    type="button"
                    variant="secondary"
          />
</div>,
      ],
      "className": "my3 px4 flex items-center clearfix",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <NavLink
              activeClassName="active"
              aria-current="page"
              className="ml4 link-text"
              exact={true}
              to="/location"
>
              Location
</NavLink>,
            <NavLink
              activeClassName="active"
              aria-current="page"
              className="ml4 link-text"
              exact={true}
              to="/contact"
>
              Contact
</NavLink>,
          ],
          "className": "col col-4 flex justify-start undefined",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "function",
            "props": Object {
              "activeClassName": "active",
              "aria-current": "page",
              "children": "Location",
              "className": "ml4 link-text",
              "exact": true,
              "to": "/location",
            },
            "ref": null,
            "rendered": "Location",
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "function",
            "props": Object {
              "activeClassName": "active",
              "aria-current": "page",
              "children": "Contact",
              "className": "ml4 link-text",
              "exact": true,
              "to": "/contact",
            },
            "ref": null,
            "rendered": "Contact",
            "type": [Function],
          },
        ],
        "type": "div",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": <NavLink
            activeClassName="active"
            aria-current="page"
            className="justify-center"
            exact={true}
            to="/"
>
            <Image
                        alt="Click the Ample Hills Logo to return to the homepage"
                        bg={false}
                        className="w100"
                        onImgLoad={[Function]}
                        src="/assets/images/ample-hills-logo.svg"
                        style={Object {}}
            />
</NavLink>,
          "className": "col col-4 flex justify-center items-center",
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "activeClassName": "active",
            "aria-current": "page",
            "children": <Image
              alt="Click the Ample Hills Logo to return to the homepage"
              bg={false}
              className="w100"
              onImgLoad={[Function]}
              src="/assets/images/ample-hills-logo.svg"
              style={Object {}}
/>,
            "className": "justify-center",
            "exact": true,
            "to": "/",
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "alt": "Click the Ample Hills Logo to return to the homepage",
              "bg": false,
              "children": null,
              "className": "w100",
              "onImgLoad": [Function],
              "src": "/assets/images/ample-hills-logo.svg",
              "style": Object {},
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          "type": [Function],
        },
        "type": "div",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <NavLink
              activeClassName="active"
              aria-current="page"
              className="mr4 link-text"
              exact={true}
              to="/profile"
>
              Profile
</NavLink>,
            <NavLink
              activeClassName="active"
              aria-current="page"
              className="mr4 link-text"
              exact={true}
              to="/collections"
>
              Collections
</NavLink>,
            <Button
              className=""
              color="peach"
              disabled={false}
              fullWidth={false}
              label="Shop Online"
              onClick={[Function]}
              to="/products"
              type="button"
              variant="secondary"
/>,
          ],
          "className": "col col-4 flex items-center justify-end clearfix undefined",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "function",
            "props": Object {
              "activeClassName": "active",
              "aria-current": "page",
              "children": "Profile",
              "className": "mr4 link-text",
              "exact": true,
              "to": "/profile",
            },
            "ref": null,
            "rendered": "Profile",
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "function",
            "props": Object {
              "activeClassName": "active",
              "aria-current": "page",
              "children": "Collections",
              "className": "mr4 link-text",
              "exact": true,
              "to": "/collections",
            },
            "ref": null,
            "rendered": "Collections",
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "function",
            "props": Object {
              "className": "",
              "color": "peach",
              "disabled": false,
              "fullWidth": false,
              "label": "Shop Online",
              "onClick": [Function],
              "to": "/products",
              "type": "button",
              "variant": "secondary",
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
        ],
        "type": "div",
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <div
            className="col col-4 flex justify-start undefined"
>
            <NavLink
                        activeClassName="active"
                        aria-current="page"
                        className="ml4 link-text"
                        exact={true}
                        to="/location"
            >
                        Location
            </NavLink>
            <NavLink
                        activeClassName="active"
                        aria-current="page"
                        className="ml4 link-text"
                        exact={true}
                        to="/contact"
            >
                        Contact
            </NavLink>
</div>,
          <div
            className="col col-4 flex justify-center items-center"
>
            <NavLink
                        activeClassName="active"
                        aria-current="page"
                        className="justify-center"
                        exact={true}
                        to="/"
            >
                        <Image
                                    alt="Click the Ample Hills Logo to return to the homepage"
                                    bg={false}
                                    className="w100"
                                    onImgLoad={[Function]}
                                    src="/assets/images/ample-hills-logo.svg"
                                    style={Object {}}
                        />
            </NavLink>
</div>,
          <div
            className="col col-4 flex items-center justify-end clearfix undefined"
>
            <NavLink
                        activeClassName="active"
                        aria-current="page"
                        className="mr4 link-text"
                        exact={true}
                        to="/profile"
            >
                        Profile
            </NavLink>
            <NavLink
                        activeClassName="active"
                        aria-current="page"
                        className="mr4 link-text"
                        exact={true}
                        to="/collections"
            >
                        Collections
            </NavLink>
            <Button
                        className=""
                        color="peach"
                        disabled={false}
                        fullWidth={false}
                        label="Shop Online"
                        onClick={[Function]}
                        to="/products"
                        type="button"
                        variant="secondary"
            />
</div>,
        ],
        "className": "my3 px4 flex items-center clearfix",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <NavLink
                activeClassName="active"
                aria-current="page"
                className="ml4 link-text"
                exact={true}
                to="/location"
>
                Location
</NavLink>,
              <NavLink
                activeClassName="active"
                aria-current="page"
                className="ml4 link-text"
                exact={true}
                to="/contact"
>
                Contact
</NavLink>,
            ],
            "className": "col col-4 flex justify-start undefined",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "function",
              "props": Object {
                "activeClassName": "active",
                "aria-current": "page",
                "children": "Location",
                "className": "ml4 link-text",
                "exact": true,
                "to": "/location",
              },
              "ref": null,
              "rendered": "Location",
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "function",
              "props": Object {
                "activeClassName": "active",
                "aria-current": "page",
                "children": "Contact",
                "className": "ml4 link-text",
                "exact": true,
                "to": "/contact",
              },
              "ref": null,
              "rendered": "Contact",
              "type": [Function],
            },
          ],
          "type": "div",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": <NavLink
              activeClassName="active"
              aria-current="page"
              className="justify-center"
              exact={true}
              to="/"
>
              <Image
                            alt="Click the Ample Hills Logo to return to the homepage"
                            bg={false}
                            className="w100"
                            onImgLoad={[Function]}
                            src="/assets/images/ample-hills-logo.svg"
                            style={Object {}}
              />
</NavLink>,
            "className": "col col-4 flex justify-center items-center",
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "function",
            "props": Object {
              "activeClassName": "active",
              "aria-current": "page",
              "children": <Image
                alt="Click the Ample Hills Logo to return to the homepage"
                bg={false}
                className="w100"
                onImgLoad={[Function]}
                src="/assets/images/ample-hills-logo.svg"
                style={Object {}}
/>,
              "className": "justify-center",
              "exact": true,
              "to": "/",
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "alt": "Click the Ample Hills Logo to return to the homepage",
                "bg": false,
                "children": null,
                "className": "w100",
                "onImgLoad": [Function],
                "src": "/assets/images/ample-hills-logo.svg",
                "style": Object {},
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
            "type": [Function],
          },
          "type": "div",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <NavLink
                activeClassName="active"
                aria-current="page"
                className="mr4 link-text"
                exact={true}
                to="/profile"
>
                Profile
</NavLink>,
              <NavLink
                activeClassName="active"
                aria-current="page"
                className="mr4 link-text"
                exact={true}
                to="/collections"
>
                Collections
</NavLink>,
              <Button
                className=""
                color="peach"
                disabled={false}
                fullWidth={false}
                label="Shop Online"
                onClick={[Function]}
                to="/products"
                type="button"
                variant="secondary"
/>,
            ],
            "className": "col col-4 flex items-center justify-end clearfix undefined",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "function",
              "props": Object {
                "activeClassName": "active",
                "aria-current": "page",
                "children": "Profile",
                "className": "mr4 link-text",
                "exact": true,
                "to": "/profile",
              },
              "ref": null,
              "rendered": "Profile",
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "function",
              "props": Object {
                "activeClassName": "active",
                "aria-current": "page",
                "children": "Collections",
                "className": "mr4 link-text",
                "exact": true,
                "to": "/collections",
              },
              "ref": null,
              "rendered": "Collections",
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "function",
              "props": Object {
                "className": "",
                "color": "peach",
                "disabled": false,
                "fullWidth": false,
                "label": "Shop Online",
                "onClick": [Function],
                "to": "/products",
                "type": "button",
                "variant": "secondary",
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
          ],
          "type": "div",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
